wordpress-mobile/WordPress-Android
# Login Rework: Enable SmartLock for Passwords in new login flow

This PR enables the user of Google's SmartLock for Passwords in the new Login flows.

The user can save their WPCOM username+password pair in SmartLock and use it again later if logging into WPCOM is needed.

This PR enables saving the credentials only when using the username/password flow.

| Native Prompt for saved credentials |
| --- |
| ![prompt-for-saved-creds](https://user-images.githubusercontent.com/1032913/28621416-8a980d60-7219-11e7-9d38-970c1b1ea700.jpg) |

To test:
* With the app data cleaned, launch the app and hit "Log in"
* Tap on the "Log in to your site by entering your site address" secondary button
* Enter the site address *of your WPCOM* site. It's probably something like "<username>.wordpress.com" and hit "Next"
* Enter your username and password and hit "Next"
* Upon successful login, a popup will ask you to optionally save your credentials to SmartLock. Do so by tapping on "SAVE PASSWORD".
* At this point, the epilogue screen is visible. Ignore it and go clean the app data again to start over.
* Launch the app and hit the "Log in" button. A popup appears asking to choose a saved WordPress account from SmartLock. There must be at least one saved account, the one we saved at a previous step. Choose that account.
* The app shows the username+password screen in the background while attempting to login with the saved credentials. It should succeed and jump to the epilogue screen.
